src/emit/mod.rs

```diff
@@ -3285,6 +3351,37 @@ pub trait TargetRuntime<'a> {
             BasicBlock { bb, phis }
         }
 
+        fn get_ins_loc(ins: &Instr) -> Option<pt::Loc> {
+            match ins {
+                Instr::Set { loc, expr, .. } => match loc {
+                    pt::Loc::File(_, _, _) => Some(*loc),
+                    _ => Some(expr.loc()),
+                },
+                Instr::Call { args, .. } if args.is_empty() => None,
```

Comment:
The `Instr` really needs a `loc` field in every possible enumeration.

Reply:
If every `Instr` is going to have a Location, please implement the trait `CodeLocation` to retrieve the `Loc`.